src/Common/Core/Impl/Extensions/CultureExtensions.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+
+namespace Microsoft.Common.Core {
+    public static class CultureExtensions {
+        private static readonly char[] _separators = new char[] { ',', ' ' };
+
+        public static string LanguageNameFromLCID(int lcid) {
+            var culture = CultureInfo.GetCultures(CultureTypes.InstalledWin32Cultures).FirstOrDefault(x => x.LCID == lcid);
+            return culture.GetLanguageName();
+        }
+
+        public static IEnumerable<string> GetLanguageNames() {
+            IEnumerable<CultureInfo> cultures = CultureInfo.GetCultures(CultureTypes.InstalledWin32Cultures);
+            return cultures.Select(c => c.GetLanguageName()).Distinct().OrderBy(x => x);
```

Comment:
Language name seemed to be friendlier than .1251. I am not sure how many are familiar with Windows LCID numbers. Actual code page is 866 but it is not unique since several other cultures use 866 as OEM. So maybe do have both?